### ISSUE TITLE
Add cue when trying to move non-grabbed balloon

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#275]
In an email, @emily-phet said

> Is there any way to know if a person has focus on the balloon, and is trying to use the “move” keys (e.g., WASD or arrow keys)? I’m wondering if we can recognize this likely situation where the user has forgotten to pick up the balloon, and show the visual cues at this point (and/or provide an alert that reminds to pick up balloon)?

In email discussion, we agreed this was a good idea, but discovered it would be difficult to determine if a screen reader user was intentionally using the "move" keys on the non-grabbed balloon to read other content in the DOM.

However, if NOT using a screen reader, we do know when the user is focused on the non-grabbed balloon and trying to move it.

@terracoda said
> If the user is using a screen reader, they will get a very similar text-based hint.
If they are on the Grab Yellow Balloon button and they advance along in the PDOM with the Down Arrow key, they will get the on-demand help text for the balloon, which says,

>"Get ready to move balloon. Once grabbed, press W, A, S, or D key to move up, left, down, or right. Space to release."

>I adjusted the wording to be more inline with the visual cue. Before it was:

>"Get ready to drag balloon. Once grabbed, press W, A, S, or D key to drag up, left, down, or right. To let go, press Space.”

@emily-phet @terracoda is there anything else to work on regarding this discussion?

**Comment by terracoda** [comment]
@jessegreenberg and @emily-phet, I'll just add that the hints are slightly different, but can be delivered at the same time regardless of technology being used.

**For non-visual users using a screen reader and keyboard** they hear that they are on the "Grab Yellow Balloon" button, and they will likely know intuitively from previous experience with web pages that they can press Space or Enter to activate the button. Therefore, the on-demand help text in the non-visual experience does not explicitly tell the user to press Space or Enter, but rather works more like an interaction hint. It hints that it might be useful to "grab" the balloon and then provides information about the custom interaction that is to come, once the balloon is grabbed.

**Visual users using a keyboard without a screen reader** receive neither the structural information "button" nor the label text describing the interaction. They just see they have focus on the Yellow Balloon. They cannot tell visually that the balloon is a button and may try to use the Arrow keys to move it. Before the balloon is actually grabbed, the Arrow keys won't work.

I like the idea of providing the visual hint on how to Grab and release the balloon, when and if a user presses an Arrow key while keyboard focus is on the Grab Yellow Balloon button, rather than directly on focus. It makes the visual hint more subtle. It would be useful to know if visual keyboard users can successfully guess on how to grab the balloon.

**Comment by terracoda** [comment]
@emily-phet, we could also provide the visual cue if a user presses any of the WASD keys while their keyboard focus is on the Grab button.

I'm not sure we could provide an audible alert (accessible screen reader users) because the letter keys and the Arrow keys are reserved for other screen reader functions. Screen reader users do get other information (as explained) from the content in and structure of the PDOM.

The current interaction of grabbing seems to be working fine for the screen readers users we have interviewed.

I think it is worth exploring providing the visual cue only after the user unsuccessfully tries moving the balloon before it is grabbed.

We could discuss other on-demand help text that is more explicit about using Space for grabbing and releasing, such as:

- Space to grab and release balloon. Once grabbed, press W, A, S, or D to move balloon up, left, down, or right.

OR

**Comment by terracoda** [comment]
Visual design of the visual hint is in #206.

**Comment by terracoda** [comment]
@emily-phet, what do you think? Do you think further changes to the wording of the on-demand help text is warranted? 

It could be useful for new screen reader users, or screen reader users who do not understand what to do with, "Grab Yellow Balloon, button".

**Comment by jessegreenberg** [comment]
@terracoda @emily-phet what is the status of this issue?  It is still relevant with the latest design doc?

**Comment by terracoda** [comment]
@emily-phet, without user testing with keyboard users, I don't think we need to adjust what we have implemented, i.e., showing the visual cue on focus, and the screen help text on-demand.

**Comment by terracoda** [comment]
@jessegreenberg, the idea behind this issue is not reflected in design documents.

**Comment by terracoda** [comment]
@emily-phet, I think this issue won't be resolved until we get feedback from visual keyboard users. We now have a visual cue for grabbing the balloon for visual users. Once we get feedback on that, we may need to iterate.

I am marking it as on-hold and removing my assignment.

**Comment by emily-phet** [comment]
Sounds good to me, I’m closing... we’ll likely just start a new issue stating the specific problem we’ve encountered if there is an issue from users.
